store/bsdb/group.go

```diff
@@ -24,15 +25,29 @@ func (b *BsDBImpl) GetGroupsByGroupIDAndAccount(groupIDList []common.Hash, accou
 		}
 	}()
 
+	// In the "group" table, each group has an account ID of "0x0000000000000000000000000000000000000000" representing the group's creation information.
+	// If this group is granted permissions, the non-existent account 0x0000000000000000000000000000000000000000 will gain access, so it's prohibited here
+	if account.String() == common.HexToAddress("0").String() {
+		return nil, nil
+	}
+
 	if includeRemoved {
+		// filter those groups already be deleted while group-accounts are still exist
+		err = b.db.Table((&Group{}).TableName()).
+			Select("group_id").
+			Where("group_id in (?) and account_id = ? and removed = false", groupIDList, account, common.HexToAddress("0")).
+			Find(&groupIDs).Error
+		if err != nil {
+			return nil, err
+		}
 		err = b.db.Table((&Group{}).TableName()).
 			Select("*").
-			Where("group_id in (?) and account_id = ?", groupIDList, account).
+			Where("group_id in (?) and account_id = ?", groupIDs, account).
 			Find(&groups).Error
 	} else {
 		err = b.db.Table((&Group{}).TableName()).
 			Select("*").
-			Where("group_id in (?) and account_id = ? and removed =false", groupIDList, account).
+			Where("group_id in (?) and account_id = ? and removed = false", groupIDList, account).
```

Comment:
Should this branch exclude "common.HexToAddress("0"))" as well?

Reply:
fixed